app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/item/InterceptClickable.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversations.messages.item
+
+import androidx.compose.foundation.Indication
+import androidx.compose.foundation.gestures.awaitEachGesture
+import androidx.compose.foundation.gestures.awaitFirstDown
+import androidx.compose.foundation.gestures.waitForUpOrCancellation
+import androidx.compose.foundation.indication
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.PressInteraction
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.pointer.PointerEventPass
+import androidx.compose.ui.input.pointer.PointerEventTimeoutCancellationException
+import androidx.compose.ui.input.pointer.pointerInput
+
+fun Modifier.interceptCombinedClickable(
+    interactionSource: MutableInteractionSource,
+    indication: Indication?,
+    onClick: (() -> Unit)? = null,
+    onLongPress: (() -> Unit)? = null,
+) = this
+    .indication(interactionSource, indication)
+    .pointerInput(Unit) {
+        awaitEachGesture {
+            val pass = PointerEventPass.Initial
+            val longPressTimeout = viewConfiguration.longPressTimeoutMillis
+
+            // wait for the first down press
+            val downEvent = awaitFirstDown(pass = pass)
+            val press = PressInteraction.Press(downEvent.position)
+
+            if (onClick != null) {
+                // if click needs to be intercepted then consume the down event
```

Comment:
What if it has both Click and Long Click?

Normal click interaction should only be triggered after `up`.

I think we should call `waitForUpOrCancellation` in any case, and measure the time (if it was shorter os longer than `longPressTimeout`).

Reply:
Here it's not about triggering click, it's just about consuming the event so that it doesn't reach children and cause multiple triggers of the same event.
If it has both `onClick` and `onLongClick` then it should consume the event anyway because it means that children shouldn't get this event at all. 
Actual triggering the click is on the line 85 and it's done exactly as you explained, `waitForUpOrCancellation` is called in any case and if it has both `onClick` and `onLongClick` then it will make use of the `longPressTimeout` to decide which one should be triggered.